packages/types/src/transaction/schema.ts

```diff
@@ -48,3 +48,10 @@ export const WasmBuildSchema = z.object({
     transactionParameters: z.unknown(),
   }),
 });
+
+export const WasmActionSchema = z.object({
+  action: z.object({
+    value: z.unknown(),
+    case: z.string(),
+  }),
+});
```

Comment:
this is how actions actually come out of wasm, a 'record' where the keyname is of ActionCase. i don't know how to get zod to understand the union type

```suggestion
//type ActionCase = string & Action['action']['case'];
export const WasmActionSchema = z.record(z.unknown());
```

Reply:
ping @grod220

Reply:
Zod can do unions: https://zod.dev/?id=unions. However, the spirit of this effort isn't so much to re-type everything (and have to keep it updated as protobufs change), but have some base type validation to provide some safety from the wasm code from sending back something completely different and rugging the frontend. So in short, it's your call if you want to enumerate all of these or just expect an object.